core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentPropertiesAndSchemaHolder.java

```diff
@@ -204,6 +204,9 @@ public void invalidate(AbsoluteTableIdentifier absoluteTableIdentifier) {
           .getTableUniqueName()
           .equals(absoluteTableIdentifier.getCarbonTableIdentifier().getTableUniqueName())) {
         SegmentIdAndSegmentPropertiesIndexWrapper value = entry.getValue();
+        synchronized (deleteIndexToDeleteTimestampMapping) {
```

Comment:
I guess in this method also we might need to add to deleteMap and check the timetolive like it is done in the below method